src/test/java/BaseTest.java

```diff
@@ -1,11 +1,269 @@
+    }
+
+    public static void enterEmail(String email){
+        WebElement emailInput = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type = 'email']")));
+        emailInput.clear();
+        emailInput.sendKeys(email);
+    }
+
+    public static void enterPassword(String password) {
+        WebElement passwordInput = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type = 'password']")));
+        passwordInput.clear();
+        passwordInput.sendKeys(password);
+    }
+
+    public static void clickSubmit(){
+        WebElement submitBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[type = 'submit']")));
+        submitBtn.click();
+    }
+
+    public void clickPlayNext(){
+        WebElement playNextBtn = driver.findElement(By.cssSelector("[title='Play next song']"));
+        playNextBtn.click();
+    }
+
+    public void clickPlay(){
+        WebElement playBtn = driver.findElement(By.cssSelector("span.play"));
+        playBtn.click();
+    }
+
+    public Boolean isSongPlay() {
+        WebElement soundBar = driver.findElement(By.cssSelector("div.bars"));
+        return soundBar.isDisplayed();
+    }
+
+    public static void openPlaylist () throws InterruptedException {
+        WebElement accessPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("a[href='#!/playlist/65568']")));
+        accessPlaylist.click();
+    }
+    public static void deleteItem () throws InterruptedException{
+        WebElement item = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".del")));
+        item.click();
+    }
+    public static void deleteBtn () throws InterruptedException {
+        WebElement btn = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("button.ok")));
+        btn.click();
+    }
+
+    public String getDeleteMsg() {
+        WebElement deleteMsg = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));
+        return deleteMsg.getText();
+    }
+
+    public static void openOption () throws InterruptedException{
+        WebElement option = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("a[href='#!/playlist/65758']")));
+        actions.contextClick(option).perform();
+    }
+
+    public static void clickEdit () throws InterruptedException{
+        WebElement edit = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("#playlists > ul > li:nth-child(3) > nav > ul > li:nth-child(1)")));
+        edit.click();
+    }
+
+    public static void rename(String playlistName) {
+        WebElement naming = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        naming.sendKeys(Keys.chord(Keys.COMMAND, "A", Keys.BACK_SPACE));
+        naming.sendKeys(playlistName);
+        naming.sendKeys(Keys.RETURN);
+    }
```

Comment:
**Page Object Model (POM):** recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

**Move these methods to the corresponding class according to the POM.**